src/afl-fuzz-init.c

```diff
@@ -653,17 +653,14 @@ void read_foreign_testcases(afl_state_t *afl, int first) {
         u32 len = write_to_testcase(afl, (void **)&mem, st.st_size, 1);
         fault = fuzz_run_target(afl, &afl->fsrv, afl->fsrv.exec_tmout);
         afl->syncing_party = foreign_name;
+        afl->foreign_file = nl[i]->d_name;
```

Comment:
This is a use after free, right? `nl[i]` is freed in line 592 above

Reply:
ah good catch!

Reply:
fix in [bedb38e](https://github.com/AFLplusplus/AFLplusplus/pull/2425/commits/bedb38e2168c5a6abddd765ea354fb6e44604479)